### ISSUE TITLE
508-defect-2 [FOCUS MANAGEMENT]: Both forms need to announce themselves consistently to screen readers and not have a negative tabindex.

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#26577]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement

As a screen reader user, I need the intercept and feedback modals to announce themselves properly and set focus inside the window so I can fill them out correctly or exit.

## Details

Axe browser plugin is reporting a **Serious** error that iFrames must not have a negative tabindex. The frames have a tabindex of -1 and I believe this is causing an issue where JAWS doesn't set virtual cursor focus on the buttons inside the frames initially. The forms are not being announced properly either.

The best thing I can recommend is to include an H1 inside the frame, and set focus on that element on first load. This will take some testing and require adjustment to the iFrame code. The best example of an accessible iFrame I could find was this one:

* https://websemantics.uk/articles/accessible-modal-dialog-popup-iframe/
* https://codepen.io/2kool2/pen/LkaXay

The code is five years old, but is vanilla JavaScript and easy to reason about.

This issue **could** become a defect level 1 if it starts throwing axe errors in our daily accessibility scans. Not doing so currently.

## Acceptance Criteria

- [ ] iFrames throw 0 axe violations
- [ ] Forms are announced to screen readers properly
- [ ] Forms have an H1 to receive focus and announce the "big idea"
- [ ] Buttons are available to receive virtual cursor focus from JAWS on first load

## WCAG or Vendor Guidance (optional)

* https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html
* https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-sequence.html
